File: src/main/java/com/example/ssauc/user/login/service/SmsService.java
package com.example.ssauc.user.login.service;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseAuthException;
import com.google.firebase.auth.FirebaseToken;
import org.springframework.stereotype.Service;

@Service
public class SmsService {

    public boolean verifyFirebaseToken(String firebaseToken, String phone) {
        try {
            FirebaseToken token = FirebaseAuth.getInstance().verifyIdToken(firebaseToken);
            String verifiedPhone = (String) token.getClaims().get("phone_number");
            return phone.equals(verifiedPhone);
        } catch (FirebaseAuthException e) {
            e.printStackTrace();
            return false;
        }
    }
}

File: src/main/java/com/example/ssauc/user/login/service/SmsService.kt
package com.example.ssauc.user.login.service

import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseAuthException
import org.springframework.stereotype.Service

@Service
class SmsService {
    fun verifyFirebaseToken(firebaseToken: String?, phone: String): Boolean {
        try {
            val token = FirebaseAuth.getInstance().verifyIdToken(firebaseToken)
            val verifiedPhone = token.claims["phone_number"] as String?
            return phone == verifiedPhone
        } catch (e: FirebaseAuthException) {
            e.printStackTrace()
            return false
        }
    }
}
